linux/src/threadpool_linux.c

```diff
@@ -291,6 +308,46 @@ static int reallocate_threadpool_array(THREADPOOL* threadpool)
     return result;
 }
 
+static TIMER_INSTANCE* get_next_timer_instance(THREADPOOL* threadpool)
+{
+    TIMER_INSTANCE* result;
+
+    // Make sure only one thread access the loop at once
+    do
+    {
+        int32_t current_val = interlocked_compare_exchange(&threadpool->timer_instance_access, 1, 0);
+        if (current_val == 0)
+        {
+            break;
+        }
+        else
+        {
+            // Do Nothing wait for address
+        }
+        (void)wait_on_address(&threadpool->timer_instance_access, current_val, UINT32_MAX);
+    } while (true);
+
+    // Loop through the list and find the first disabled timer
+    int32_t index;
+    for (index = 0; index < MAX_TIMER_INSTANCE_COUNT; index++)
+    {
+        if (interlocked_add(&threadpool->timer_instance[index].timer_status, 0) == TIMER_DISABLED)
+        {
+            result = &threadpool->timer_instance[index];
+            break;
+        }
+    }
```

Comment:
This doesn't work though if two calls to get_next_timer_instance come at the same time.

The first call would find index 0 under the "lock", then the next thread could also find index 0.

Seems like you need to change the timer_status under this "lock".